examples/slash_commands/main.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"flag"
+	dgo "github.com/bwmarrin/discordgo"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+var (
+	botToken  = flag.String("bot_tok", "", "Bot token")
+	testGuild = flag.String("test_guild", "", "Guild for testing the bot")
+)
+
+func setup() *dgo.Session {
+	client, err := dgo.New("Bot " + *botToken)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	client.AddHandler(func(*dgo.Session, *dgo.Ready) {
+		log.Println("Bot is up!")
+	})
+
+	return client
+}
+
+func main() {
+	flag.Parse()
+
+	client := setup()
+
+	client.AddHandler(func(s *dgo.Session, i *dgo.InteractionCreate) {
+		responseType := dgo.InteractionResponseType(i.Interaction.Data.Options[0].Value.(float64))
+		responseData := &dgo.InteractionApplicationCommandResponseData{
+			TTS:     false,
+			Content: "here we go",
+			// Flags: 1 << 6,
```

Comment:
Why the commented code and what does `1 << 6` represent?

Reply:
Message like Clyde's - hidden.

Reply:
My point is that the intent of this code is not immediately obvious to the user of this library which examples should be.